src/Facebook/InstantArticles/Elements/Caption.php

```diff
@@ -345,62 +317,60 @@ public function getPosition()
      * @see Caption::VERTICAL_BOTTOM
      * @see Caption::VERTICAL_CENTER
      */
-    public function getVerticalAlignment()
+    public function getVerticalAlignment(): string
     {
         return $this->verticalAlignment;
     }
 
     /**
-     * Structure and create the full ArticleImage in a XML format DOMElement.
+     * Structure and create the full ArticleImage in a XML format DOMNode.
      *
      * @param \DOMDocument $document where this element will be appended. Optional
-     * @return \DOMElement
+     * @return \DOMNode
      */
-    public function toDOMElement($document = null)
+    public function toDOMElement(\DOMDocument $document): \DOMNode
```

Comment:
`$document` was optional before. Should it stay this way?
I think `toDOMElement(\DOMDocument $document = new \DOMDocument())` would have the same semantics.

Reply:
Technically the main issue of us rendering the elements and creating by ourselves the DOMDocument, is that we will miss the correct encoding to write content to.
I feel that keeping it optional (so we create it all the time) works well only for test purposes, but for practical usage of our lib, we should drop this being optional.

What do you think?

Reply:
Sounds good